webserver/pkgpkr/webservice/tests/test_views.py

```diff
@@ -190,4 +194,94 @@ def test_recommendations_json(self):
         response = recommendations_json(request, 'pkgpkr1/express')
         self.assertEqual(response.status_code, 200)
         self.assertIsNotNone(response.content)
-        self.assertIn('"current_branch": "test"', response.content.decode())
\ No newline at end of file
+        self.assertIn('"current_branch": "test"', response.content.decode())
+
+    def assertion_helper_for_recom_service_api(self, post_data, expected_code, expected_message_substring):
+
+        request = self.factory.post('/api/recommendations',
+                                    data=post_data,
+                                    content_type='application/json')
+
+        response = recommendations_service_api(request)
+        self.assertEqual(response.status_code, expected_code)
+        self.assertIn(expected_message_substring, response.content.decode())
+
+        return response
+
+    def test_recommendations_service_api(self):
+        # Assert Method not found
+        request = self.prep_not_github_auth_request('/api/recommendations', instantiate_user_session=False)
+        request.method = 'GET'
+        response = recommendations_service_api(request)
+        self.assertEqual(response.status_code, 405)
+
+        ### Assert issues ###
+
+
+        # Issue with json, NOTE: there's no way to pass bad JSON, so just changing content_type so it's not there
+        post_data = "THIS IS NOT A JSON!"
+        self.assertion_helper_for_recom_service_api(post_data, 500, 'Could not parse JSON')
+
+        # Issue with missing keys
+        post_data = {}
+        self.assertion_helper_for_recom_service_api(post_data, 500, 'Required JSON key')
+
+        # Issue with language value
+        post_data = {"language": "NOT SUPPORTED VALUE", "dependencies": {"lodash": "v4.17.15"}}
```

Comment:
fwiw, the `v` in front of the version would make this fail anyway, I think. Should probably remove that if that's not what you're testing here.